### PR TITLE
ci(fix): replace abandoned actions-rs

### PR DESCRIPTION
**Pull request by leet4tari**
Description
Replace ```actions-rs``` with actively developed actions.  

Motivation and Context
Fix warnings in GitHub Action workflows - https://github.com/tari-project/tari/issues/5361

How Has This Been Tested?
Some parts have been tested in local fork.

What process can a PR reviewer use to test or verify this change?
Check actions working as expected and no warnings produced. 

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

**Review by stringhandler — DISMISSED** [review]

**Comment by AaronFeickert** [comment]
Would [`rust-toolchain`](https://github.com/dtolnay/rust-toolchain) do the trick as well? It's much more broadly used and written by a developer on the Rust team, so perhaps it's a better long-term choice if feasible to use.

**Comment by AaronFeickert** [comment]
Looks like you're requesting the `stable` toolchain when loading the `rust-toolchain` action, which won't match the toolchain you later specify. The [documentation](https://github.com/dtolnay/rust-toolchain#inputs) for the action has advice for how to handle this properly.

**Review by AaronFeickert — CHANGES_REQUESTED** [review]
Looks like you're requesting the `stable` toolchain when loading the `rust-toolchain` action, which won't match the toolchain you later specify. The [documentation](https://github.com/dtolnay/rust-toolchain#inputs) for the action has advice for how to handle this properly.

**Comment by leet4tari** [comment]
> Looks like you're requesting the `stable` toolchain when loading the `rust-toolchain` action
```
dtolnay/rust-toolchain@stable
```
is the stable tag/branch of the ```dtolnay/rust-toolchain@stable```
ie:
```
        uses: dtolnay/rust-toolchain@stable
        with:
          components: rustfmt, clippy
          toolchain: ${{ matrix.builds.rust }}
```
```toolchain``` would be ```nightly-2022-11-03```, thou we might be able to drop that too and just be using the ```rust-toolchain.toml``` default.

**Comment by AaronFeickert** [comment]
Isn't the `stable` tag intended to correspond with the `stable` toolchain? Or am I misunderstanding the nature of that correspondence?

At any rate, the [documentation](https://github.com/dtolnay/rust-toolchain#inputs) recommends:
> When passing an explicit toolchain as an input instead of `@rev`, you'll want to use "dtolnay/rust-toolchain@master" as the revision of the action.

**Comment by leet4tari** [comment]
Updated, let's see how it runs

**Comment by AaronFeickert** [comment]
Looks like a `cargo check` [error](https://github.com/tari-project/tari/actions/runs/5145512657/jobs/9263254077?pr=5401#step:8:28) is present.

**Comment by leet4tari** [comment]
Need to replace all the ```actions-rs``` org actions, including the ```actions-rs/cargo```, thou I think we have another problem to look into, which might me related to ```error: usage of sparse registries requires "-Z sparse-registry"```

**Comment by leet4tari** [comment]
Should be ready for merge, builds all work in my own fork.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,109 +35,80 @@ jobs:
...
         with:
           toolchain: ${{ env.toolchain }}
-          profile: minimal
-          override: true
       - name: ubuntu dependencies
         run: |
           sudo apt-get update
           sudo bash scripts/install_ubuntu_dependencies.sh
+      - name: rustup show
+        run: rustup show
```
Comment (AaronFeickert) [inline .github/workflows/ci.yml:86]: Is this just for human-readable debugging output of some kind?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -34,109 +35,80 @@ jobs:
...
-          override: true
       - uses: Swatinem/rust-cache@v2
       - name: ubuntu dependencies
         run: |
           sudo apt-get update
           sudo bash scripts/install_ubuntu_dependencies.sh
       - name: rustup show
-        run: |
-          rustup show
+        run: rustup show
```
Comment (AaronFeickert) [inline .github/workflows/ci.yml:109]: See other comment.

Comment (leet4tari) [reply .github/workflows/ci.yml:109]: was there before, I think it's for confirming what rust deps are at build.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (leet4tari) [reply .github/workflows/ci.yml:86]: Match a good idea just to confirm what rust deps are in place at build time.

**Review by SWvheerden — APPROVED** [review]

**Pull request merged**